servlet/src/main/java/io/undertow/servlet/api/DeploymentInfo.java

```diff
@@ -121,6 +121,7 @@
     private boolean sendCustomReasonPhraseOnError = false;
     private boolean useCachedAuthenticationMechanism = true;
     private boolean preservePathOnForward = true;
+    private boolean allowOrphanSession = false;
```

Comment:
I think the default value should be `true` for compatibility reasons at least during EAP 7.x releases. (I guess it's too late to add a new configuration option to the undertow subsystem for EAP 7.x as it requires the management model change.)
We may be able to change this to `false` for future releases (or set this option to `false` by default in the undertow subsystem layer since EAP 8.x).

Reply:
@msfm This proposal is not targeting an existing branch.

Reply:
@msfm By design, Undertow should set its defaults to conform to the servlet specification.  Any container integrating Undertow (e.g. WildFly) is free to set its own default value.

Reply:
> This proposal is not targeting an existing branch.
>
> By design, Undertow should set its defaults to conform to the servlet specification. Any container integrating Undertow (e.g. WildFly) is free to set its own default value.

I agree. 

My concern was happening to merge the same change into the undertow branches for EAP 7.x. Of course, I understand that the WildFly/EAP integration layer (undertow subsystem layer) can set its own default value. However, I thought having a different default value may cause confusion. 

Anyway, as this change is not targeting existing branches but for future releases, I'm fine with `false` by default.